core/src/main/codegen/templates/Parser.jj

```diff
@@ -6065,10 +6065,22 @@ SqlCall GroupByWindowingCall():
     final List<SqlNode> args;
 }
 {
-    <TUMBLE> { s = span(); }
-    args = UnquantifiedFunctionParameterList(ExprContext.ACCEPT_SUB_QUERY) {
-        return SqlStdOperatorTable.TUMBLE.createCall(s.end(this), args);
-    }
+    (
+        <TUMBLE> { s = span(); }
+        args = UnquantifiedFunctionParameterList(ExprContext.ACCEPT_SUB_QUERY) {
+            return SqlStdOperatorTable.TUMBLE.createCall(s.end(this), args);
+        }
+    |
+        <HOP> { s = span(); }
+        args = UnquantifiedFunctionParameterList(ExprContext.ACCEPT_SUB_QUERY) {
+            return SqlStdOperatorTable.HOP.createCall(s.end(this), args);
+        }
+    |
+        <SESSION> { s = span(); }
+        args = UnquantifiedFunctionParameterList(ExprContext.ACCEPT_SUB_QUERY) {
+            return SqlStdOperatorTable.SESSION.createCall(s.end(this), args);
+        }
```

Comment:
We can combine these
```java
args = UnquantifiedFunctionParameterList(ExprContext.ACCEPT_SUB_QUERY) {
            return SqlStdOperatorTable.SESSION.createCall(s.end(this), args);
        }
```
Use another variable to keep the operator and move the common parse logic to the java code block.

Reply:
Done.

Reply:
`span()` can also be promoted.

Reply:
I tried to change to this way
```
(
        <TUMBLE>
        {
            op = SqlStdOperatorTable.TUMBLE_OLD;
        }
    |
        <HOP>
        {
            op = SqlStdOperatorTable.HOP_OLD;
        }
    |
        <SESSION>
        {
            op = SqlStdOperatorTable.SESSION_OLD;
        }
    )
    args = UnquantifiedFunctionParameterList(ExprContext.ACCEPT_SUB_QUERY) {
        s = span();
        return op.createCall(s.end(this), args);
    }
```
But the parser didn't work as expected.

Any thing I have missed?

Reply:
No worries, i would promote that when i merged the PR.